src/foam/nanos/auth/PasswordExpiryAuthService.js

```diff
@@ -22,34 +22,39 @@ foam.CLASS({
   javaImports: [
     'foam.dao.DAO',
     'foam.nanos.NanoService',
-    'static foam.mlang.MLang.EQ'
+
+    'static foam.mlang.MLang.AND',
+    'static foam.mlang.MLang.EQ',
+    'static foam.mlang.MLang.OR'
   ],
 
   methods: [
     {
       name: 'start',
       javaCode:
-`if ( getDelegate() instanceof NanoService ) {
-  ((NanoService) getDelegate()).start();
-}`
+        `if ( getDelegate() instanceof NanoService ) {
+          ((NanoService) getDelegate()).start();
+        }`
     },
     {
       name: 'login',
       javaCode:
-`User user = (User) ((DAO) getLocalUserDAO()).find(userId);
-if ( isPasswordExpired(user) ) {
-  throw new AuthenticationException("Password expired");
-}
-return getDelegate().login(x, userId, password);`
-    },
-    {
-      name: 'loginByEmail',
-      javaCode:
-`User user = (User) ((DAO) getLocalUserDAO()).inX(x).find(EQ(User.EMAIL, email.toLowerCase()));
-if ( isPasswordExpired(user) ) {
-  throw new AuthenticationException("Password expired");
-}
-return getDelegate().loginByEmail(x, email, password);`
+        `User user = (User) ((DAO) getLocalUserDAO())
+          .inX(x)
+          .find(
+            AND(
+              OR(
+                EQ(foam.nanos.auth.User.EMAIL, id.toLowerCase()),
+                EQ(foam.nanos.auth.User.USER_NAME, id)
+              ),
+              EQ(foam.nanos.auth.User.LOGIN_ENABLED, true)
```

Comment:
Should we be checking any of these properties on User as well?

* enabled (deprecated)
* emailVerified
* status
* deleted (will be deprecated soon)
* lifecycleState (nanopay side)

Reply:
emailVerified: AND ( EQ(email, id), EQ(emailverified, true)
status: this is compliance - don't know if we block login on compliance issues - no way for them to correct anything - please inquire with Nadir. 
lifecycle: EQ (active)

Reply:
enabled: I will remove this.

emailVerified: We currently allow users with an unverified email to login to Ablii. They just get stuck at the "We sent you and email, please verify you email" page, so I don't think we should add the emailVerified condition. 

status: Checked in with Nadir and he said that we should prevent a user from logging in if their status is set to "Disabled" or "Revoked". So AND(NEQ(User.Status, "Revoked"), NEQ(User.Status, "Disabled"))

lifecycle: add this.